internal/services/services.go

```diff
@@ -77,5 +89,73 @@ func toggleServices(
 			}
 		}
 	}
+
+	waitGroup.Wait()
 	return nil
 }
+
+func listenToAutoPortForwardingChangesOnRemote(
+	ctx context.Context,
+	serviceName string,
+	w io.Writer,
+	projectDir string,
+	action serviceAction,
+	waitGroup *sync.WaitGroup,
+) error {
+
+	if os.Getenv("DEVBOX_REGION") == "" {
+		return nil
+	}
+	hostname, err := hostname()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	fmt.Fprintf(w, "Waiting for port forwarding to start/stop for service %q\n", serviceName)
+	waitGroup.Add(1)
+
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Can the cancel of this context be deferred? It ultimately needs to be called everywhere otherwise it'll leak resources.